src/services/homework/hooks/submissions.js

```diff
@@ -181,90 +216,126 @@ const insertSubmissionData = (context) => {
 		const homeworkService = context.app.service('/homework');
-		return homeworkService.get(homeworkId, {
-			account: { userId },
-			query: {
-				$populate: ['courseId'],
-			},
-		})
+		return homeworkService
+			.get(homeworkId, {
+				account: { userId },
+				query: {
+					$populate: ['courseId'],
+				},
+			})
 			.then((homework) => {
 				context.data.homework = homework;
 				// isTeacher?
 				context.data.isTeacher = false;
-				if ((equalIds(context.data.homework.teacherId, userId))
-					|| (context.data.homework.courseId.teacherIds || [])
-						.some((teacherId) => equalIds(teacherId, userId))
-					|| (context.data.homework.courseId.substitutionIds || [])
-						.some((subsId) => equalIds(subsId, userId))
+				if (
+					equalIds(context.data.homework.teacherId, userId) ||
+					(
+						context.data.homework.courseId.teacherIds || []
+					).some((teacherId) => equalIds(teacherId, userId)) ||
+					(
+						context.data.homework.courseId.substitutionIds || []
+					).some((subsId) => equalIds(subsId, userId))
 				) {
 					context.data.isTeacher = true;
 				}
 				return Promise.resolve(context);
 			})
-			.catch(() => Promise.reject(new GeneralError({ message: "can't reach homework service" })));
+			.catch(() =>
+				Promise.reject(
+					new GeneralError({
+						message: "can't reach homework service",
+					}),
+				),
+			);
 	}
 	return Promise.reject(new BadRequest());
 };
 
 const insertSubmissionsData = (context) => {
-// get all the submissions for the homework
+	// get all the submissions for the homework
 	const submissionService = context.app.service('/submissions');
-	return submissionService.find({
-		query: {
-			homeworkId: context.data.homeworkId,
-			$select: ['teamMembers'],
-			$populate: [{ path: 'studentId', select: ['_id'] }],
-		},
-	}).then((submissions) => {
-		context.data.submissions = submissions.data;
-		return Promise.resolve(context);
-	})
-		.catch(() => Promise.reject(new GeneralError({ message: "can't reach submission service" })));
+	return submissionService
+		.find({
+			query: {
+				homeworkId: context.data.homeworkId,
+				$select: ['teamMembers'],
+				$populate: [{ path: 'studentId', select: ['_id'] }],
+			},
+		})
+		.then((submissions) => {
+			context.data.submissions = submissions.data;
+			return Promise.resolve(context);
+		})
+		.catch(() =>
+			Promise.reject(
+				new GeneralError({ message: "can't reach submission service" }),
+			),
+		);
 };
 
 const preventNoTeamMember = (context) => {
 	if (!(context.data.submission || {}).teamMembers) {
 		context.data.teamMembers = [context.params.account.userId];
 	}
-	if (context.method == 'update' && (!context.data.teamMembers || context.data.teamMembers.length == 0)) {
-		return Promise.reject(new Conflict({
-			message: 'Abgaben ohne TeamMember sind nicht erlaubt!',
-		}));
+	if (
+		context.method == 'update' &&
+		(!context.data.teamMembers || context.data.teamMembers.length == 0)
```

Comment:
Codacy found an issue: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=5987920)